src/renderers/dom/shared/ReactDOMEventListener.js

```diff
@@ -178,6 +179,17 @@ var ReactDOMEventListener = {
     var targetInst = ReactDOMComponentTree.getClosestInstanceFromNode(
       nativeEventTarget,
     );
+    if (
+      targetInst !== null &&
+      typeof targetInst.tag === 'number' &&
+      !ReactFiberTreeReflection.isFiberMounted(targetInst)
+    ) {
+      // If we get an event (ex: img onload) before committing that
+      // component's mount, ignore it for now (that is, treat it as if it was an
+      // event on a non-React tree). We might also consider queueing events and
+      // dispatching them after the mount.
+      targetInst = null;
+    }
```

Comment:
This doesn't block the event if something down below the tree is already mounted since we're not doing this at every level of handleTopLevelImpl's ancestor algorithm. There are mostly theoretical cases where this could matter:

1) A non-Fiber tree is nested and shares the event system. Not going to happen with our current set up and we'll probably delete Stack by then.

2) A React subtree node gets manually inserted inside of a not-yet mounted tree. This probably doesn't happen other than for SSR hydration since you can't get access to the newly created nodes. Maybe a custom element.

Another case that can happen is that we block something further up the tree that shouldn't be. I think that can actually happen with hydration:

```js
// sync
ReactDOM.render(<div onClick={handleClick}><div ref={n => leaf = n} dangerouslySetInnerHTML={ssr} /></div>, container);
// async, hydration
ReactDOM.render(<Foo />, leaf);
```

It'll block all events to the parent while the sub-tree is hydrating.